File: app/src/main/java/com/mandarin/bcu/androidutil/stage/adapters/EnemyListRecycle.java
package com.mandarin.bcu.androidutil.stage.adapters;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;
import android.support.v4.view.ViewCompat;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;

import com.mandarin.bcu.R;

import common.util.stage.Stage;

public class EnemyListRecycle extends RecyclerView.Adapter<EnemyListRecycle.ViewHolder> {
    private final Activity activity;
    private final Stage st;
    private int multi;

    public EnemyListRecycle(Activity activity, Stage st) {
        this.activity = activity;
        this.st = st;
    }

    public EnemyListRecycle(Activity activity, Stage st, int multi) {
        this.activity = activity;
        this.st = st;
        this.multi = multi;
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        RecyclerView listView;
        Button frse;

        ViewHolder(@NonNull View row) {
            super(row);

            listView = row.findViewById(R.id.stginfoenemlist);
            frse = row.findViewById(R.id.stginfoenfrse);
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View row = LayoutInflater.from(activity).inflate(R.layout.stage_enemy_layout,viewGroup,false);

        return new ViewHolder(row);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        SharedPreferences shared = activity.getSharedPreferences("configuration", Context.MODE_PRIVATE);

        viewHolder.listView.setLayoutManager(new LinearLayoutManager(activity));
        ViewCompat.setNestedScrollingEnabled(viewHolder.listView,false);

        StEnListRecycle listAdapter = new StEnListRecycle(activity,st,multi,shared.getBoolean("frame",true));

        viewHolder.listView.setAdapter(listAdapter);

        if(!shared.getBoolean("frame",true))
            viewHolder.frse.setText(activity.getString(R.string.config_seconds));

        viewHolder.frse.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(viewHolder.frse.getText().toString().equals(activity.getString(R.string.config_frames))) {
                    StEnListRecycle listAdapter = new StEnListRecycle(activity,st,multi,false);

                    viewHolder.listView.setAdapter(listAdapter);

                    viewHolder.frse.setText(activity.getString(R.string.config_seconds));
                } else {
                    StEnListRecycle listAdapter = new StEnListRecycle(activity,st,multi,true);

                    viewHolder.listView.setAdapter(listAdapter);

                    viewHolder.frse.setText(activity.getString(R.string.config_frames));
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return 1;
    }
}

File: app/src/main/java/com/mandarin/bcu/androidutil/stage/adapters/EnemyListRecycle.kt
package com.mandarin.bcu.androidutil.stage.adapters

import android.app.Activity
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.core.view.ViewCompat
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.mandarin.bcu.R
import com.mandarin.bcu.androidutil.StaticStore
import common.util.stage.Stage

open class EnemyListRecycle : RecyclerView.Adapter<EnemyListRecycle.ViewHolder> {
    private val activity: Activity
    private val st: Stage
    private var multi = 0
    private val mapcode: Int
    private val custom: Boolean

    constructor(activity: Activity, st: Stage, mapcode: Int, custom: Boolean) {
        this.activity = activity
        this.st = st
        this.mapcode = mapcode
        this.custom = custom
    }

    constructor(activity: Activity, st: Stage, multi: Int, mapcode: Int, custom: Boolean) {
        this.activity = activity
        this.st = st
        this.multi = multi
        this.mapcode = mapcode
        this.custom = custom
    }

    inner class ViewHolder(row: View) : RecyclerView.ViewHolder(row) {
        var listView: RecyclerView = row.findViewById(R.id.stginfoenemlist)
        var frse: Button = row.findViewById(R.id.stginfoenfrse)

    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val row = LayoutInflater.from(activity).inflate(R.layout.stage_enemy_layout, viewGroup, false)
        return ViewHolder(row)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val shared = activity.getSharedPreferences(StaticStore.CONFIG, Context.MODE_PRIVATE)

        viewHolder.listView.layoutManager = LinearLayoutManager(activity)

        ViewCompat.setNestedScrollingEnabled(viewHolder.listView, false)

        val listAdapter = StEnListRecycle(activity, st, multi, shared.getBoolean("frame", true),mapcode, custom)

        viewHolder.listView.adapter = listAdapter

        if (!shared.getBoolean("frame", true))
            viewHolder.frse.text = activity.getString(R.string.config_seconds)

        viewHolder.frse.setOnClickListener {
            if (viewHolder.frse.text.toString() == activity.getString(R.string.config_frames)) {
                val listAdapter1 = StEnListRecycle(activity, st, multi, false, mapcode, custom)
                viewHolder.listView.adapter = listAdapter1
                viewHolder.frse.text = activity.getString(R.string.config_seconds)
            } else {
                val listAdapter2 = StEnListRecycle(activity, st, multi, true, mapcode, custom)
                viewHolder.listView.adapter = listAdapter2
                viewHolder.frse.text = activity.getString(R.string.config_frames)
            }
        }
    }

    override fun getItemCount(): Int {
        return 1
    }
}